lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
There's more to it than just knowing context. Additional information about *where* the icon will be displayed needs to be provided. In my case, I need to know if the target is a directory, a symlink, or already has existing classes applied. [You can see for yourself here](https://github.com/DanBrooker/file-icons/blob/29218e42fbdbd9d877ec9eb5c3a76848f282e016/lib/service/icon-service.coffee#L146).